nipype/pipeline/engine/test_newengine/test_single_node.py

```diff
@@ -0,0 +1,103 @@
+from supernode import Node # or whatever
+
+import numpy as np
+
+import pytest, pdb
+
+def fun1(a, **dict):
+    return a**2
+
+def fun2(a, **dict):
+    pow = np.arange(4)
+    return a**pow
+
+def fun3(a, b, **dict):
+    return a * b
+
+
+
+@pytest.mark.parametrize("inputs_dict, expected_output", [ 
+        ({"a": [3, 4, 5]}, [9, 16, 25]),
+        ({"a": [[3, 4, 5], [6, 7, 8]]}, [[9, 16, 25], [36, 49, 64]]),
```

Comment:
whereas this would work:

`({"a": [np.array([3, 4, 5]), np.array([6, 7, 8])]}, [np.array([9, 16, 25]), np.array([36, 49, 64])]),`